config/src/lib.rs

```diff
@@ -824,7 +824,11 @@ impl Config {
     /// returns the [`ethers_solc::ConfigurableArtifacts`] for this config, that includes the
     /// `extra_output` fields
     pub fn configured_artifacts_handler(&self) -> ConfigurableArtifacts {
-        ConfigurableArtifacts::new(self.extra_output.clone(), self.extra_output_files.clone())
+        let mut extra_output_files = self.extra_output_files.clone();
+        if !extra_output_files.contains(&ContractOutputSelection::Metadata) {
+            extra_output_files.push(ContractOutputSelection::Metadata);
+        }
```

Comment:
lgtm, but let's add a rationale for this

Reply:
fixed